rwkv.cpp

```diff
@@ -1364,12 +1360,46 @@ bool rwkv_eval_sequence(const struct rwkv_context * ctx, const uint32_t * sequen
     return true;
 }
 
-uint32_t rwkv_get_state_buffer_element_count(const struct rwkv_context * ctx) {
-    return ctx->instance->model.header.n_layer * 5 * ctx->instance->model.header.n_embed;
+// Provided for compatibility
+extern "C" RWKV_API uint32_t rwkv_get_state_buffer_element_count(const struct rwkv_context * ctx) {
+    return rwkv_state_len(ctx);
 }
 
-uint32_t rwkv_get_logits_buffer_element_count(const struct rwkv_context * ctx) {
-    return ctx->instance->model.header.n_vocab;
+// Provided for compatibility
+extern "C" RWKV_API uint32_t rwkv_get_logits_buffer_element_count(const struct rwkv_context * ctx) {
+    return rwkv_logits_len(ctx);
+}
+
+size_t rwkv_get_n_vocab(const struct rwkv_context * ctx) {
+    return (size_t) ctx->instance->model.header.n_vocab;
+}
+
+size_t rwkv_get_n_embed(const struct rwkv_context * ctx) {
+    return (size_t) ctx->instance->model.header.n_embed;
+}
+
+size_t rwkv_get_n_layer(const struct rwkv_context * ctx) {
+    return (size_t) ctx->instance->model.header.n_layer;
+}
+
+size_t rwkv_state_len(const struct rwkv_context * ctx) {
```

Comment:
```suggestion
size_t rwkv_get_state_len(const struct rwkv_context * ctx) {
```
For consistency with other getters. Same for `rwkv_get_logits_len`